python/BioSimSpace/Sandpit/Exscientia/FreeEnergy/_absolute.py

```diff
@@ -0,0 +1,202 @@
+######################################################################
+# BioSimSpace: Making biomolecular simulation a breeze!
+#
+# Copyright: 2017-2023
+#
+# Authors: Lester Hedges <lester.hedges@gmail.com>
+#
+# BioSimSpace is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# BioSimSpace is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with BioSimSpace. If not, see <http://www.gnu.org/licenses/>.
+#####################################################################
+
+"""
+Functionality for absolute free-energy simulations. Extremely similar relative,
+but adds functionality for dealing with receptor-ligand restraints.
+"""
+
+__all__ = ["Absolute", "getData"]
+
+import os as _os
+import sys as _sys
+
+from sire.legacy.Base import getBinDir as _getBinDir
+from sire.legacy.Base import getShareDir as _getShareDir
+
+from .._Exceptions import MissingSoftwareError as _MissingSoftwareError
+from ._restraint import Restraint as _Restraint
+from ._relative import Relative as _Relative
+
+# Check that the analyse_freenrg script exists.
+if _sys.platform != "win32":
+    _analyse_freenrg = _os.path.join(_getBinDir(), "analyse_freenrg")
+else:
+    _analyse_freenrg = _os.path.join(
+        _os.path.normpath(_getShareDir()), "scripts", "analyse_freenrg.py"
+    )
+if not _os.path.isfile(_analyse_freenrg):
+    raise _MissingSoftwareError(
+        "Cannot find free energy analysis script in expected location: '%s'"
+        % _analyse_freenrg
+    )
+if _sys.platform == "win32":
+    _analyse_freenrg = "%s %s" % (
+        _os.path.join(_os.path.normpath(_getBinDir()), "sire_python.exe"),
+        _analyse_freenrg,
+    )
+
+
+class Absolute(_Relative):
```

Comment:
If you don't like the name of _Relative, would it be better to have a class called alchemy? It seems that `Relative` is s strict subclass of this `Absolute` class?

Reply:
I agree that this isn't the best way to do things. Shall I implement @lohedges's suggestion and:

1) Move all functionality to an `AlchemicalFreeEnergy` base class
2) Derive `Relative` and `Absolute` classes which are currently both identical to the base class, but allow for slightly different functionality in each in future?

This would also avoid having to update the tutorials to use the new naming and would avoid braking any current code.

Reply:
Also, do you have any comments on the above @lohedges?

Reply:
My understand of class is that say if you have two `set` of functionality A and B.
The base class will be `A.intersection(B)`.
The class A will be `A - B`.
The class B will be `B - A`.

This set up makes sense if both `A - B` and `B - A` are not empty.
However, in this case, `Relative` is a strict subset of `Absolute`, so there is no functionally in `Relative` that is not used by `Absolute`.

I think it might be better to keep them in the same class and only separate when there is need to do that. In my experience, creating new classes this too early will make the later refactor very difficult.

For tutorials, I think you could just create a dummy class `Absolute(Relative)` while Relative still retain the restraint related functionality?

Reply:
I agree that creating separate classes doesn't make sense currently, but thought it might be helpful to prevent API changes when/ if we end up needing to separate ABFE/RBFE. However, if it's better in your experience, I'm very happy to keep everything in a single class. If both you and @lohedges agree, I'll move everything back to the same class and rename it `AlchemicalFreeEnergy`.

Reply:
I think that this is something that we can revisit at a later date when the functionality is moved out of the sandpit. My main concern was that you either have confusion with the naming of the class, or if there are attributes of the `Absolute` class that aren't used, or don't make sense if you aren't doing a `Relative` simulation. At present it is probably best that the implementation is consistent with what is expected in the sandpit, so I'm happy with what ever @xiki-tempula would prefer. As mentioned, we could always temporarily alias the class name for the purposes of the tutorial if it makes things clearer.